SVG: Move declaration of templated methods to inline header to fix non-unified builds

csaavedra opened the pull request:
#### 0e84668159728e97372768600e1768c04e19e90e
<pre>
SVG: Move declaration of templated methods to inline header to fix non-unified builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=296178">https://bugs.webkit.org/show_bug.cgi?id=296178</a>

Reviewed by Rob Buis.

With non-unified builds, the implementation of both
parseAndReplaceTransform() and parseTransform() don&apos;t get
instantiated for all the template types they are used with,
which results in a linking error. This can be fixed by moving
the implementation of those methods to an inline header file that
can then be included where appropriate, so that the methods are
instantiated in those compilation units as expected.

This linking error doesn&apos;t happen with unified builds, as
all the related SVG files get bundled into one compilation unit.

* Source/WebCore/svg/SVGTransformList.cpp:
* Source/WebCore/svg/SVGTransformable.cpp:
(WebCore::parseTransformParamList): Deleted.
(WebCore::SVGTransformable::parseAndReplaceTransform): Deleted.
(WebCore::SVGTransformable::parseTransform): Deleted.
* Source/WebCore/svg/SVGTransformable.h:
* Source/WebCore/svg/SVGTransformableInlines.h: Copied from Source/WebCore/svg/SVGTransformable.cpp.
(WebCore::parseTransformParamList):
(WebCore::SVGTransformable::parseAndReplaceTransform):
(WebCore::SVGTransformable::parseTransform):

Canonical link: <a href="https://commits.webkit.org/297595@main">https://commits.webkit.org/297595@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eab2ef32421618ced5994bcc1c310fc46d5cb67b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31981 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22459 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118337 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Running checkout-pull-request; Running compile-webkit") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62587 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114212 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32634 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40545 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85283 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/35963 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115197 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26040 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100998 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65713 "Build is in progress. Recent messages:Printed configuration; Checked out pull request; run-api-tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25344 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19134 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62174 "Built successfully") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95424 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Running apply-patch; Checked out pull request; run-api-tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19213 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121654 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39324 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29260 "Passed tests") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94090 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode; 25 flakes 69 failures; Uploaded test results; 13 flakes 50 failures; Compiled WebKit; 3 flakes 47 failures; Passed layout tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39705 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97240 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93914 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24013 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39149 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16943 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/35353 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39212 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44700 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Running checkout-pull-request; Found 71156 issues") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38847 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42184 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40590 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/eab2ef32421618ced5994bcc1c310fc46d5cb67b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48238

rwlbuis reviewed (approved):
LGTM.

csaavedra committed:
SVG: Move declaration of templated methods to inline header to fix non-unified builds
https://bugs.webkit.org/show_bug.cgi?id=296178

Reviewed by Rob Buis.

With non-unified builds, the implementation of both
parseAndReplaceTransform() and parseTransform() don't get
instantiated for all the template types they are used with,
which results in a linking error. This can be fixed by moving
the implementation of those methods to an inline header file that
can then be included where appropriate, so that the methods are
instantiated in those compilation units as expected.

This linking error doesn't happen with unified builds, as
all the related SVG files get bundled into one compilation unit.

* Source/WebCore/svg/SVGTransformList.cpp:
* Source/WebCore/svg/SVGTransformable.cpp:
(WebCore::parseTransformParamList): Deleted.
(WebCore::SVGTransformable::parseAndReplaceTransform): Deleted.
(WebCore::SVGTransformable::parseTransform): Deleted.
* Source/WebCore/svg/SVGTransformable.h:
* Source/WebCore/svg/SVGTransformableInlines.h: Copied from Source/WebCore/svg/SVGTransformable.cpp.
(WebCore::parseTransformParamList):
(WebCore::SVGTransformable::parseAndReplaceTransform):
(WebCore::SVGTransformable::parseTransform):

Canonical link: https://commits.webkit.org/297595@main
